data-service/src/component-tests/java/uk/gov/gchq/palisade/component/data/model/AuditableAuthorisedDataRequestTest.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2018-2021 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package uk.gov.gchq.palisade.component.data.model;
+
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.json.JsonTest;
+import org.springframework.boot.test.json.JacksonTester;
+import org.springframework.boot.test.json.JsonContent;
+import org.springframework.boot.test.json.ObjectContent;
+import org.springframework.test.context.ContextConfiguration;
+
+import uk.gov.gchq.palisade.service.data.model.AuditableAuthorisedDataRequest;
+
+import java.io.IOException;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertAll;
+import static uk.gov.gchq.palisade.component.data.common.CommonTestData.AUDITABLE_DATA_REQUEST;
+import static uk.gov.gchq.palisade.component.data.common.CommonTestData.AUDITABLE_DATA_REQUEST_WITH_ERROR;
+
+@JsonTest
+@ContextConfiguration(classes = {AuditableAuthorisedDataRequestTest.class})
+class AuditableAuthorisedDataRequestTest {
```

Comment:
I do appreciate that they seem out pf place, but they do serve a purpose.  I want some easy way to test the methods the Builder class and its' methods.  To do this I want two instances of the same object and compare them.  If for example I forget to set an instance variable, forget to override the equal method , the will code will compile, but the test will fail   The lazy part is that we already have this code that uses Json for serialising/deserialising.  I'm using it as a mean to an end by making the second instance of the object that I can use in the assertion.

Reply:
Tests can act as a kind of documentation, demonstrating expected usage of the code.
If this is a test of the builders etc. then the test methods and testing mechanisms should be named appropriately.
Otherwise, I don't think we need tests on what is effectively a POJO

Reply:
Good point.  I have adjusted the test so it uses the builder to create the second instance instead of using Json.